test/e2e/pipeline/infra/cluster-api/cni.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+#shellcheck disable=SC1091
+
+FILEPATH=$(realpath "$0")
+WORKDIR=$(dirname "$FILEPATH")
+
+# shellcheck source=./pre-requirements.sh
+source "$WORKDIR/pre-requirements.sh"
+
+function install_calico() {
+    local kubeconfig=$1
+    curl https://raw.githubusercontent.com/projectcalico/calico/v3.24.4/manifests/calico.yaml |
+        sed -E 's|^( +)# (- name: CALICO_IPV4POOL_CIDR)$|\1\2|g;'"\
+s|^( +)# (  value: )\"192.168.0.0/16\"|\1\2\"$POD_CIDR\"|g;"'/- name: CLUSTER_TYPE/{ n; s/( +value: ").+/\1k8s"/g };''/- name: CALICO_IPV4POOL_IPIP/{ n; s/value: "Always"/value: "Never"/ };''/- name: CALICO_IPV4POOL_VXLAN/{ n; s/value: "Never"/value: "Always"/};''/# Set Felix endpoint to host default action to ACCEPT./a\            - name: FELIX_VXLANPORT\n              value: "6789"' |
+        "${KUBECTL}" apply -f - --kubeconfig "$kubeconfig"
+}
+
+function install_cilium() {
+    local kubeconfig=$1
+
+    if [ ! -f "${BINDIR/cilium/}" ]; then
+        setup_arch_and_os
+        local CILIUM_CLI_VERSION
+        CILIUM_CLI_VERSION=$(curl -s https://raw.githubusercontent.com/cilium/cilium-cli/master/stable.txt)
```

Comment:
I don't know if it is better to use the latest tag to detect possible issues with new Cilium versions or to have a static version set to have a more stable testing scenario

Reply:
Better to set a astatic version 👍